src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java

```diff
@@ -207,22 +206,72 @@ public void stopContainer(String containerName) {
             String[] auxiliaryRepositoriesNames, Path buildScriptPath) {
-        copyToContainer(assignmentRepositoryPath.toString(), buildJobContainerId);
-        renameDirectoryOrFile(buildJobContainerId, assignmentRepositoryPath.getFileName().toString(), "assignment-repository");
-        copyToContainer(testRepositoryPath.toString(), buildJobContainerId);
-        renameDirectoryOrFile(buildJobContainerId, testRepositoryPath.getFileName().toString(), "test-repository");
+
+        ExecCreateCmdResponse createDirectoryCmdReponse = dockerClient.execCreateCmd(buildJobContainerId).withCmd("mkdir", "/repositories").exec();
+        dockerClient.execStartCmd(createDirectoryCmdReponse.getId()).exec(new ResultCallback.Adapter<>());
+
+        addAndPrepareDirectory(buildJobContainerId, testRepositoryPath, "repositories/test-repository", true);
+        addAndPrepareDirectory(buildJobContainerId, assignmentRepositoryPath, "repositories/assignment-repository", true);
+
+        // also need the assignment repository within test-repository
+        addAndPrepareDirectory(buildJobContainerId, assignmentRepositoryPath, "repositories/test-repository/assignment", true);
 
         for (int i = 0; i < auxiliaryRepositoriesPaths.length; i++) {
-            copyToContainer(auxiliaryRepositoriesPaths[i].toString(), buildJobContainerId);
-            renameDirectoryOrFile(buildJobContainerId, auxiliaryRepositoriesPaths[i].getFileName().toString(), auxiliaryRepositoriesNames[i] + "-repository");
+            addAndPrepareDirectory(buildJobContainerId, auxiliaryRepositoriesPaths[i], "repositories/test-repository/" + auxiliaryRepositoriesNames[i] + "/", true);
+        }
+
+        addAndPrepareDirectory(buildJobContainerId, buildScriptPath, "script.sh", false);
+    }
+
+    private void addAndPrepareDirectory(String containerId, Path repositoryPath, String newDirectoryName, Boolean isDirectory) {
+        copyToContainer(repositoryPath.toString(), containerId);
+        renameDirectoryOrFile(containerId, repositoryPath.getFileName().toString(), newDirectoryName);
+        if (isDirectory) {
+            convertDosFilesToUnix(newDirectoryName + "/", containerId);
+        }
+        else {
+            convertDosFilesToUnix(newDirectoryName, containerId);
         }
-        copyToContainer(buildScriptPath.toString(), buildJobContainerId);
-        renameDirectoryOrFile(buildJobContainerId, buildScriptPath.getFileName().toString(), "script.sh");
     }
 
     private void renameDirectoryOrFile(String containerId, String oldName, String newName) {
         ExecCreateCmdResponse renameDirectoryCmdResponse = dockerClient.execCreateCmd(containerId).withCmd("mv", oldName, newName).exec();
-        dockerClient.execStartCmd(renameDirectoryCmdResponse.getId()).exec(new ResultCallback.Adapter<>());
+        final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
the latch definition is duplicated, maybe you could generalize it to duplicate the code a bit.

Reply:
Moved the docker command execution to a method.